tempodb/wal/wal.go

```diff
@@ -73,27 +73,35 @@ func New(c *Config) (*WAL, error) {
 	}, nil
 }
 
-func (w *WAL) AllBlocks() ([]*ReplayBlock, error) {
+func (w *WAL) AllBlocks() ([]*AppendBlock, []error, error) { // jpe test failure modes. delete file, warnings
```

Comment:
AllBlocks previously was a fairly light-weight operation, but is now quite intense (parsing entire contents of all wal files before returning). This design also prevents ability to log progress well from the ingester when replaying.   Maybe instead it could return metadata or a new kind of wal-specific info, and the ingester can call  `newAppendBlockFromFile` or an equivalent new thing, and log progress again.

Reply:
Passed in a logger for visibilty and renamed.